### ISSUE TITLE
Problem Using PiZiGate on Hass.io

### ISSUE THREAD

**Issue by doudz** [doudz/homeassistant-zigate#49]

**Comment by cobch** [comment]
Ok. Sorry. It started with my problem pairing Ikea plug but that is true it should be more in hassio and PiZigate. Thanks.

**Comment by doudz** [comment]
So you confirm you've added the following in config.txt ?
`dtoverlay=pi3-miniuart-bt`

**Comment by doudz** [comment]
Could you make a screenshot of your `Hassio - System - Hardware` ?

**Comment by cobch** [comment]
Yes. I have try to add the following at the end of my config.txt:
`dtoverlay=pi3-miniuart-bt`
and 
```
enable_uart=1
dtoverlay=pi3-miniuart-bt
```

Here is a screenshot of my hass.io:
![IMG_7017](https://user-images.githubusercontent.com/50397676/57715180-6426e580-7676-11e9-9d3f-67ee4d1e0a30.jpg)

**Comment by doudz** [comment]
Thanks
It's not the screenshot I expected.
Please click on "**HASSIO**" (side menu) then **System** tab and **Hardware** button

**Comment by cobch** [comment]
![IMG_7018](https://user-images.githubusercontent.com/50397676/57717927-8459a300-767c-11e9-8dab-2e3741e9c2c3.jpg)

**Comment by doudz** [comment]
so /dev/ttyAMA0 is here.
The problem is probably on GPIO side, I'll take a look

**Comment by doudz** [comment]
Just a question, beside the file config.txt, is there cmdline.txt file ?

**Comment by cobch** [comment]
Yes there is one. See below all the files I have on my hassos boot:
![IMG_7019](https://user-images.githubusercontent.com/50397676/57727693-7020a080-7692-11e9-8c57-74b4de424b32.jpg)

I have one more problem, I don't know if it is related but each time I reboot my hassio the reboot is unsuccesfull. I have to remove the power cable and then reboot the system by plug in the power cable again. See picture below where the system get stuck during a reboot.
![IMG_7010](https://user-images.githubusercontent.com/50397676/57727852-cd1c5680-7692-11e9-807d-97a139e28ef4.jpg)

An alternative is to call the shut down service of my Hassio and then remove and plug again my power cable. One thing I have noticed is that the blue light is always in strong blue even when the Hassio is shut down.

I am not a prof and don't know whether my observations are usefull to fix the problem.

**Comment by doudz** [comment]
Could you give the content of cmdline.txt ?

I don't know about the reboot problem. Hassio is really specific, I prefer and use raspbian

**Comment by doudz** [comment]
I see in your log that you use deconz, could you try without ?

I agree you need to put the following in config.txt
```
enable_uart=1
dtoverlay=pi3-miniuart-bt
```

**Comment by cobch** [comment]
I give you the content of cmdline.txt tonight. I will also try to remove Deconz. 

My problem with rebooting hassio arrived when I installed PiZigate, that is why I think it may be connected to it.

Keep in touch!

**Comment by doudz** [comment]
The reboot problem is probably because the zigate lib is hanging trying to connect

**Comment by cobch** [comment]
My cmdline.txt is:
`dwc_otg.lpm_enable=0 console=tty1`

I have tried to start without Deconz but same problem "No response after command 0x0049 (14)"

**Comment by cobch** [comment]
I wonder if we should not configure the name of the zigate as mentionned in the raspbee tutorial ( "null" -> "dev/ttyAMA0")?
![IMG_7024](https://user-images.githubusercontent.com/50397676/57788528-87fe3000-7737-11e9-9c4a-95b35648b2f8.jpg)
![IMG_7023](https://user-images.githubusercontent.com/50397676/57788536-8c2a4d80-7737-11e9-8303-84d28f77897b.jpg)

**Comment by doudz** [comment]
Could try to replace `dtoverlay=pi3-miniuart-bt` by `dtoverlay=pi3-disable-bt` in config.txt ?

**Comment by cobch** [comment]
Still same problem. "No response after command 0x0049 (9)"
Can I make a reset of the PiZigate? I start to doubt that I will be able to use it with hassio )-:

**Comment by doudz** [comment]
Are you using the 32bit or 64bit Hassio image ?

**Comment by cobch** [comment]
32 because it seems more compatible

**Comment by doudz** [comment]
Well, I have no more Idea... 
Did you check your pizigate on raspbian ? It could be a firmware problem 
Or it could be related to gpio ... I don't have rpi3 available to check

**Comment by cobch** [comment]
Yes I tried and it was working to make the test from the tutorial on ZiGate.fr 

https://zigate.fr/documentation/tester-sa-pizigate/

But I did not try to use it with Home assistant on Raspbian.

**Comment by cobch** [comment]
What do you think about that:

> Before starting the add-on for the first time after installing or upgrading, in the add-on config you must specify the device name that has been assigned to your RaspBee/Conbee as the "deconz_device". Replace null and specify the device name in quotes (e.g. "/dev/ttyUSB0" or "/dev/ttyAMA0"). 

We should maybe do the same for PiZiGate?

**Comment by cobch** [comment]
How to change that within hassio?
<img width="454" alt="Screenshot 2019-05-15 at 22 31 26" src="https://user-images.githubusercontent.com/50397676/57807394-5ef29500-7761-11e9-91f7-2bf49d51793c.png">

**Comment by doudz** [comment]
> How to change that within hassio?
> <img alt="Screenshot 2019-05-15 at 22 31 26" width="454" src="https://user-images.githubusercontent.com/50397676/57807394-5ef29500-7761-11e9-91f7-2bf49d51793c.png">

No it's a different thing, in our case the null is because wy can't get IEEE from ZiGate

**Comment by doudz** [comment]
Are you using rpio_gpio component ?
I'm asking because rpio_gpio use BCM mode and zigate use BOARD mode so it could be a problem

**Comment by cobch** [comment]
How can I check that? No idea!

**Comment by doudz** [comment]
> How can I check that? No idea!

So I suppose the answer is that you don't use it https://www.home-assistant.io/components/rpi_gpio/

**Comment by cobch** [comment]
No. Nothing like that on my config. Should I return the PiZigate?

**Comment by doudz** [comment]
Could you try to modify the file `__init__.py` of the component ?
Replace
`REQUIREMENTS = ['zigate==0.29.5']`
By
`REQUIREMENTS = ['https://github.com/doudz/zigate/archive/dev.zip#1.0.0']
`

**Comment by cobch** [comment]
I have replaced 
`REQUIREMENTS = ['zigate==0.29.4']` (not the same version as above, "4" instead of "5")
by
`REQUIREMENTS = ['https://github.com/doudz/zigate/archive/dev.zip#1.0.0']`

Log after a reboot:
2019-05-17 15:06:19 WARNING (MainThread) [homeassistant.loader] You are using a custom integration for zigate which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you do experience issues with Home Assistant.
2019-05-17 15:07:00 DEBUG (SyncWorker_9) [custom_components.zigate] Port : /dev/ttyAMA0
2019-05-17 15:07:00 DEBUG (SyncWorker_9) [custom_components.zigate] Host : None
2019-05-17 15:07:00 DEBUG (SyncWorker_9) [custom_components.zigate] GPIO : True
2019-05-17 15:07:00 DEBUG (SyncWorker_9) [custom_components.zigate] Led : True
2019-05-17 15:07:00 DEBUG (SyncWorker_9) [custom_components.zigate] Channel : None
2019-05-17 15:07:01 DEBUG (SyncWorker_9) [custom_components.zigate] ZiGate object created <zigate.core.ZiGateGPIO object at 0x703201d0>
2019-05-17 15:07:07 DEBUG (SyncWorker_1) [zigate] Try loading persistent file
2019-05-17 15:07:07 DEBUG (SyncWorker_1) [zigate] Load success
2019-05-17 15:07:07 DEBUG (SyncWorker_1) [zigate] REQUEST : 0x0010 
2019-05-17 15:07:07 ERROR (ZiGate-Listen) [zigate] OOPS connection lost, reconnect...
2019-05-17 15:07:07 DEBUG (SyncWorker_1) [zigate] Msg to send b'0010000010'
2019-05-17 15:07:07 DEBUG (SyncWorker_1) [zigate] Encoded Msg to send b'01021010021002101003'
2019-05-17 15:07:07 DEBUG (SyncWorker_1) [zigate] Waiting for status message for command 0x0010
2019-05-17 15:07:07 ERROR (ZiGate-Listen) [zigate] OOPS connection lost, reconnect...
2019-05-17 15:07:10 WARNING (SyncWorker_1) [zigate] No response after command 0x0010 (1)
2019-05-17 15:07:10 WARNING (SyncWorker_1) [zigate] Failed to retrieve zigate firmware version
2019-05-17 15:07:10 DEBUG (SyncWorker_1) [zigate] REQUEST : 0x0021 b'\x03\x18\xc8\x00'
2019-05-17 15:07:10 DEBUG (SyncWorker_1) [zigate] Msg to send b'00210004f60318c800'
2019-05-17 15:07:10 DEBUG (SyncWorker_1) [zigate] Encoded Msg to send b'0102102102100214f6021318c8021003'
2019-05-17 15:07:10 DEBUG (SyncWorker_1) [zigate] Waiting for status message for command 0x0021
2019-05-17 15:07:11 ERROR (SyncWorker_17) [homeassistant.components.yeelight] Unable to update bulb status: A socket error occurred when sending the command.
2019-05-17 15:07:13 WARNING (SyncWorker_1) [zigate] No response after command 0x0021 (2)
2019-05-17 15:07:13 DEBUG (SyncWorker_1) [zigate] REQUEST : 0x0023 b'\x00'
2019-05-17 15:07:13 DEBUG (SyncWorker_1) [zigate] Msg to send b'002300012200'
2019-05-17 15:07:13 DEBUG (SyncWorker_1) [zigate] Encoded Msg to send b'010210230210021122021003'
2019-05-17 15:07:13 DEBUG (SyncWorker_1) [zigate] Waiting for status message for command 0x0023
2019-05-17 15:07:16 WARNING (SyncWorker_1) [zigate] No response after command 0x0023 (3)
2019-05-17 15:07:16 DEBUG (SyncWorker_1) [zigate] Check network state
2019-05-17 15:07:16 DEBUG (SyncWorker_1) [zigate] REQUEST : 0x0009 
2019-05-17 15:07:16 DEBUG (SyncWorker_1) [zigate] Msg to send b'0009000009'
2019-05-17 15:07:16 DEBUG (SyncWorker_1) [zigate] Encoded Msg to send b'010210021902100210021903'
2019-05-17 15:07:16 DEBUG (SyncWorker_1) [zigate] Waiting for status message for command 0x0009
2019-05-17 15:07:20 WARNING (SyncWorker_1) [zigate] No response after command 0x0009 (4)
2019-05-17 15:07:20 ERROR (SyncWorker_1) [zigate] Failed to get network state
2019-05-17 15:07:20 DEBUG (SyncWorker_1) [zigate] Network is down, start it
2019-05-17 15:07:20 DEBUG (SyncWorker_1) [zigate] REQUEST : 0x0024 
2019-05-17 15:07:20 DEBUG (SyncWorker_1) [zigate] Msg to send b'0024000024'
2019-05-17 15:07:20 DEBUG (SyncWorker_1) [zigate] Encoded Msg to send b'01021024021002102403'
2019-05-17 15:07:20 DEBUG (SyncWorker_1) [zigate] Waiting for status message for command 0x0024
2019-05-17 15:07:23 WARNING (SyncWorker_1) [zigate] No response after command 0x0024 (5)
2019-05-17 15:07:23 DEBUG (SyncWorker_1) [zigate] REQUEST : 0x0015 
2019-05-17 15:07:23 DEBUG (SyncWorker_1) [zigate] Msg to send b'0015000015'
2019-05-17 15:07:23 DEBUG (SyncWorker_1) [zigate] Encoded Msg to send b'01021015021002101503'
2019-05-17 15:07:23 DEBUG (SyncWorker_1) [zigate] Waiting for status message for command 0x0015
2019-05-17 15:07:26 WARNING (SyncWorker_1) [zigate] No response after command 0x0015 (6)
2019-05-17 15:07:26 DEBUG (SyncWorker_1) [zigate] Auto saving /config/zigate.json
2019-05-17 15:07:26 DEBUG (SyncWorker_1) [zigate] Saving persistent file
2019-05-17 15:07:26 DEBUG (SyncWorker_1) [zigate] File already existing, make a backup before
2019-05-17 15:07:26 DEBUG (SyncWorker_1) [zigate] REQUEST : 0x0018 b'\x01'
2019-05-17 15:07:26 DEBUG (SyncWorker_1) [zigate] Msg to send b'001800011801'
2019-05-17 15:07:26 DEBUG (SyncWorker_1) [zigate] Encoded Msg to send b'010210180210021118021103'
2019-05-17 15:07:26 DEBUG (SyncWorker_1) [zigate] Waiting for status message for command 0x0018
2019-05-17 15:07:29 WARNING (SyncWorker_1) [zigate] No response after command 0x0018 (7)
2019-05-17 15:07:29 DEBUG (SyncWorker_1) [zigate] REQUEST : 0x0010 
2019-05-17 15:07:29 DEBUG (SyncWorker_1) [zigate] Msg to send b'0010000010'
2019-05-17 15:07:29 DEBUG (SyncWorker_1) [zigate] Encoded Msg to send b'01021010021002101003'
2019-05-17 15:07:29 DEBUG (SyncWorker_1) [zigate] Waiting for status message for command 0x0010
2019-05-17 15:07:32 WARNING (SyncWorker_1) [zigate] No response after command 0x0010 (8)
2019-05-17 15:07:32 WARNING (SyncWorker_1) [zigate] Failed to retrieve zigate firmware version
2019-05-17 15:07:32 ERROR (MainThread) [homeassistant.core] Error doing job: Future exception was never retrieved
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/zigate/__init__.py", line 352, in start_zigate
    version = myzigate.get_version_text()
  File "/config/deps/lib/python3.7/site-packages/zigate/core.py", line 782, in get_version_text
    v = self.get_version(refresh)['version']
TypeError: 'NoneType' object is not subscriptable




I hav also noticed that I have 2 zygate.json files. See below:
![Screen Shot 2019-05-17 at 15 00 05](https://user-images.githubusercontent.com/50397676/57929790-88700580-78b4-11e9-9ad7-01901a152dc0.png)

**Comment by doudz** [comment]
zigate.json.0 is just a backup of zigate.json

No more idea...

Or :
- Check again your pizigate according to zigate.fr
- If necessary try to flash / erase EEPROM again...
- Use HA in Raspbian instead of Hassio...

**Comment by cobch** [comment]
Ok. You mean this tutorial?
https://zigate.fr/documentation/tester-sa-pizigate/

**Comment by doudz** [comment]
Yes

**Comment by cobch** [comment]
I have already tried but will test again tomorrow. I don’t want to go back to home assistant on Raspbian since I think that hassio is the best platform for automatisation and smart home. 

Could it be something that great the problem with the PiZigate?
https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

**Comment by cobch** [comment]
Hi again,
I tested the PiZigate and the test is showing me that the PiZigate is working well. I do not find instructions how to flash the PiZigate, only instructions for Zigate with usb. Any advices?

**Comment by cobch** [comment]
Any more ideas? I will send back the PiZiGate otherwise since Hassio is the native system for running Home Assistant and don't not want to change it only for Zigate (-;

**Comment by doudz** [comment]
Or if you want to wait few weeks, I will order a Rpi3 to make some tests

**Comment by cobch** [comment]
Of course! It will be really great to make it works. I wonder only how to flash or update the PiZigate. Could you send some instructions?

**Comment by doudz** [comment]
I received a Rpi3. I will now make some test with HASSIO

**Comment by cobch** [comment]
Thanks Sébastien,

I did not have the patience to wait and returned the PiZigate. But I am sure it will be needed in the futur by someone!

Best,
Charles

**Comment by doudz** [comment]
After checking, I got no problem to use pizigate with Hass.io on Rpi3
So I don't know why it wasn't working for you.

As I said you need to :
Add the following to config.txt in boot partition
```
enable_uart=1
dtoverlay=pi3-miniuart-bt
```
And add the following in configuration.yaml

```
zigate:
  gpio: true
```

So I close this issue since there's no problem

**Comment by doudz** [comment]
![image](https://user-images.githubusercontent.com/1262548/59567764-37e5f680-9072-11e9-8ea0-2012bb29ed3d.png)

**Comment by cobch** [comment]
Great to hear and so sorry about it. Sound’s stupid from my part!

Charles

**Comment by davidsp59** [comment]
> zigate:
>   gpio: true

Hello,
I have the same config (raspi3b pizigate hassio) and same problem. Can not add a zigate device.

I modified the configuration.yaml and config.txt file

With this configuration in the YAML file
zigate:
  gpio: true

unable to restart the server, the file is not good.


I tried this instead:
zha:
  usb_path: pizigate: / dev / serial0

the server restarted, no error message but no device detected


an idea to unlock me ??
